tests/testthat/test-step-join.R

```diff
@@ -154,3 +154,24 @@ test_that("performs cartesian joins as needed", {
   out <- collect(left_join(x, y, by = "x"))
   expect_equal(nrow(out), 9)
 })
+
+test_that("performs cross join", {
+  df1 <- data.frame(x = 1:2, y = "a")
```

Comment:
You need `stringsAsFactors = FALSE` here (or use a tibble) to avoid the test failures.